### PR TITLE
Start 3

### PR DESCRIPTION
**Pull request by vskurikhin**
Intermediate merge

### CODE REVIEW

File: rest-employees-service/src/main/java/su/svn/href/controllers/DepartmentsRestController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            default:
+                return departmentFullDao.findAll(offset, limit);
+        }
+    }
+
+    @GetMapping("/{id}")
+    public Mono<DepartmentDto> readDepartment(@PathVariable Long id)
+    {
+        return departmentFullDao.findById(id);
+    }
```
Comment (victorskurihin) [inline rest-employees-service/src/main/java/su/svn/href/controllers/DepartmentsRestController.java:65]: Need unit + integration tests

**Review by victorskurihin — CHANGES_REQUESTED** [review]

File: rest-employees-service/src/main/java/su/svn/href/dao/DepartmentFullDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    public static String SELECT =
+        "SELECT d.department_id, department_name, d.manager_id,"
+            + " m.first_name, m.last_name, m.email, m.phone_number,"
+            + " d.location_id, l.street_address, l.postal_code, l.city, l.state_province, l.country_id"
+            + " FROM departments d"
+            + " LEFT JOIN employees m ON m.employee_id = d.manager_id"
+            + " LEFT JOIN locations l ON l.location_id = d.location_id";
+
+    @Autowired
```
Comment (victorskurihin) [inline rest-employees-service/src/main/java/su/svn/href/dao/DepartmentFullDaoImpl.java:25]: Need integration test

File: rest-employees-service/src/main/java/su/svn/href/dao/EmployeeDao.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package su.svn.href.dao;
+
+import org.springframework.data.r2dbc.repository.query.Query;
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import su.svn.href.models.Employee;
+
+public interface EmployeeDao extends ReactiveCrudRepository<Employee, Long>
```
Comment (victorskurihin) [inline rest-employees-service/src/main/java/su/svn/href/dao/EmployeeDao.java:9]: Need unit + integration tests

File: rest-employees-service/src/main/java/su/svn/href/models/Employee.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import java.time.LocalDate;
+import java.util.Date;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+@Table("employees")
+public class Employee
```
Comment (victorskurihin) [inline rest-employees-service/src/main/java/su/svn/href/models/Employee.java:17]: Need unit tests

File: rest-employees-service/src/main/java/su/svn/href/models/dto/DepartmentDto.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    private String departmentName;
+
+    private Manager manager;
+
+    private Location location;
+
+    private Collection<Employee> employees = new LinkedList<>();
+
+    public static DepartmentDto collectFromMap(Map<String, Object> map)
```
Comment (victorskurihin) [inline rest-employees-service/src/main/java/su/svn/href/models/dto/DepartmentDto.java:32]: Very need unit tests

**Review by victorskurihin — CHANGES_REQUESTED** [review]

**Comment by vskurikhin** [comment]
Ok

**Pull request closed without merging**